### ISSUE TITLE
Segfault when Mapfile includes a missing file

### ISSUE THREAD

**Issue by ejn** [MapServer/MapServer#7016]
Given the following mapfile `test.map` where `missing.map` does not exist:

```
MAP
   include "missing.map"
END  #MAP
```

Running `map2img -m test.map -o test.jpg` produces a segfault.

Running in GDB produces the following:
```
Program received signal SIGSEGV, Segmentation fault.
msyy_load_buffer_state () at /home/bjorn/code/MapServer/maplexer.c:5028
```

This is all tested on Debian Bookworm (Mapserver 8.0.0 installed from the distro repos, so it's [this line here](https://github.com/MapServer/MapServer/blob/rel-8-0-0/maplexer.c#L5028)), but I can't see any changes in `maplexer.c` in main since then which would appear relevant, so I assume the problem is still present.

Output `map2img -v`:
```
MapServer version 8.0.0 OUTPUT=PNG OUTPUT=JPEG OUTPUT=KML SUPPORTS=PROJ SUPPORTS=AGG SUPPORTS=FREETYPE SUPPORTS=CAIRO SUPPORTS=SVG_SYMBOLS SUPPORTS=RSVG SUPPORTS=ICONV SUPPORTS=FRIBIDI SUPPORTS=WMS_SERVER SUPPORTS=WMS_CLIENT SUPPORTS=WFS_SERVER SUPPORTS=WFS_CLIENT SUPPORTS=WCS_SERVER SUPPORTS=SOS_SERVER SUPPORTS=OGCAPI_SERVER SUPPORTS=FASTCGI SUPPORTS=THREADS SUPPORTS=GEOS SUPPORTS=PBF INPUT=JPEG INPUT=POSTGIS INPUT=OGR INPUT=GDAL INPUT=SHAPEFILE INPUT=FLATGEOBUF
```

**Comment by geographika** [comment]
For info I'm unable to recreate this on Windows with the latest MapServer codebase. I get the following output:

```
map2img -m test.map -o mytest.png -conf mapserver.conf
msLoadMap(): Premature End-of-File.  <br>
msyylex(): Unable to access file. Error opening included file "missing.map". <br>
```

**Comment by ejn** [comment]
> For info I'm unable to recreate this on Windows with the latest MapServer codebase. I get the following output:
> 
> ```
> map2img -m test.map -o mytest.png -conf mapserver.conf
> msLoadMap(): Premature End-of-File.  <br>
> msyylex(): Unable to access file. Error opening included file "missing.map". <br>
> ```

Thanks for testing @geographika : Looks like I'll have to see if I can get main up and running locally to test this.

**Comment by ejn** [comment]
Obviously should have taken the time to check it out properly rather than just looking rather perfunctorily at the changelog for maplexer.c before I posted the issue: The problem was fixed in #6854

Sorry for the noise!
